modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-account-test/src/testIntegration/java/com/liferay/headless/commerce/admin/account/resource/v1_0/test/AccountChannelEntryResourceTest.java

```diff
@@ -15,15 +15,1259 @@
+				Collections.singletonMap(
+					LocaleUtil.getSiteDefault(), RandomTestUtil.randomString()),
+				RandomTestUtil.randomString(), 1000,
+				CommerceTermEntryConstants.TYPE_PAYMENT_TERMS, null,
+				serviceContext);
+
+		_commercePriceList =
+			CommercePriceListLocalServiceUtil.addCommercePriceList(
+				RandomTestUtil.randomString(), testGroup.getGroupId(),
+				_user.getUserId(), _commerceCurrency.getCommerceCurrencyId(),
+				true, CommercePriceListConstants.TYPE_PRICE_LIST, 0, true,
+				RandomTestUtil.randomString(), 1000, 1, 1, 2022, 12, 0, 0, 0, 0,
+				0, 0, true, serviceContext);
+
+		_commerceUser = UserLocalServiceUtil.addUser(
+			_user.getUserId(), testCompany.getCompanyId(), true,
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(), true,
+			RandomTestUtil.randomString(),
+			RandomTestUtil.randomString() + "@liferay.com",
+			LocaleUtil.getSiteDefault(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(), 0, 0,
+			true, 1, 1, 2022, RandomTestUtil.randomString(), null, null, null,
+			null, false, serviceContext);
+
+		Role role = RoleLocalServiceUtil.getRole(
+			testCompany.getCompanyId(), RoleConstants.ADMINISTRATOR);
+
+		UserLocalServiceUtil.addRoleUser(
+			role.getRoleId(), _commerceUser.getUserId());
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+
+		CommerceAddressLocalServiceUtil.deleteCommerceAddress(_commerceAddress);
+		CommerceCurrencyLocalServiceUtil.deleteCommerceCurrency(
+			_commerceCurrency);
+		CommerceAccountLocalServiceUtil.deleteCommerceAccount(_commerceAccount);
+		CommerceChannelLocalServiceUtil.deleteCommerceChannel(
+			_commerceChannel1);
+		CommerceChannelLocalServiceUtil.deleteCommerceChannel(
+			_commerceChannel2);
+		CommerceChannelLocalServiceUtil.deleteCommerceChannel(
+			_commerceChannel3);
+		CommerceDiscountLocalServiceUtil.deleteCommerceDiscount(
+			_commerceDiscount);
+		CommerceTermEntryLocalServiceUtil.deleteCommerceTermEntry(
+			_commerceDeliveryTerm);
+		CommerceTermEntryLocalServiceUtil.deleteCommerceTermEntry(
+			_commercePaymentTerm);
+		UserLocalServiceUtil.deleteUser(_commerceUser);
+		UserLocalServiceUtil.deleteUser(_user);
+	}
+
+	@Override
+	protected String[] getAdditionalAssertFieldNames() {
+		return new String[] {"overrideEligibility", "priority"};
+	}
+
+	@Override
+	protected AccountChannelEntry randomAccountChannelEntry() throws Exception {
+		return new AccountChannelEntry() {
+			{
+				accountExternalReferenceCode = StringUtil.toLowerCase(
+					RandomTestUtil.randomString());
+				accountId = RandomTestUtil.randomLong();
+				channelExternalReferenceCode = StringUtil.toLowerCase(
+					RandomTestUtil.randomString());
+				channelId = _getFirstAvailableChannel().getPrimaryKey();
+				classExternalReferenceCode = StringUtil.toLowerCase(
+					RandomTestUtil.randomString());
+				classPK = RandomTestUtil.randomLong();
+				id = RandomTestUtil.randomLong();
```

Comment:
I'm not sure all of these ids should be randomly created, 
- for the account the API should use the account found with the accountId from the endpoint
- classPK should use the PK of the entity that should already exists
- id should be created once the POST succeeds

So I don't understand how the tests are passing :D

Reply:
That's because 
1) the APIs are using the accountId from the endpoint to performe the operation (do not retrieve the accountId or accountExternalReferenceCode from the "object")
2) classPK is overwritten before posting because it depends on the type of the "resource" you're managing
3) id is the same of the point 1 (it's not retrieved from the "object")